Makefile

```diff
@@ -0,0 +1,33 @@
+.PHONY: setup-cachix
+setup-cachix:
+	cachix use iohk
+	cachix use mpickering
+	cachix use test-ptitfred
+
+setup-nix-build:
+	mkdir -p .nix-result
+
+.PHONY: build
+build: setup-nix-build setup-cachix
+	nix build -f ci.nix --out-link .nix-result/lib hmacaroons.components.library
+
+.PHONY: test
+test:
```

Comment:
I should make all targets that uses nix depending on `setup-cachix` and `setup-nix-build` .